tensorboard/data/proto/data_provider.proto

```diff
@@ -2,12 +2,16 @@ syntax = "proto3";
 
 package tensorboard.data;
 
+import "google/protobuf/timestamp.proto";
+
 import "tensorboard/compat/proto/summary.proto";
 import "tensorboard/compat/proto/tensor.proto";
 
 option go_package = "github.com/tensorflow/tensorboard/tensorboard/data/proto/data_provider_proto";
 
 service TensorBoardDataProvider {
+  // Get metadata about an experiment.
```

Comment:
I don't object to evolving this interface a little away from Python `DataProvider` but FWIW we should anticipate that we might want to change this further depending on the outcome of experiment-ifying Python `DataProvider` as part of upcoming work.